doc/R-vignette/igraph_extended_tutorial_R_vignette_english.Rmd

```diff
@@ -0,0 +1,621 @@
+---
+title: "Introduction to igraph"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Introduction to igraph}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+# Tutorial
+
+The main goals of the igraph library is to provide a set of data types and functions for 1) pain-free implementation of graph algorithms, 2) fast handling of large graphs, with millions of vertices and edges, 3) allowing rapid prototyping via high level languages like R.
+
+## Creating a graph
+
+The simplest way to create a graph is the `Graph` constructor. To make an empty graph:
+
+```{r}
+library("igraph")
+
+gi <- make_empty_graph()
+```
+
+To make a graph with 10 nodes (numbered `1` to `10`) and two edges connecting nodes `1-2` and `1-5`:
+
+```{r}
+gi <- graph(edges = c(1,2, 1,5), n=10, directed = FALSE)
+```
+
+We can print the graph to get a summary of its nodes and edges:
+
+```{r echo = TRUE}
+print(gi)
+```
+
+This means: **U** ndirected graph with **10** vertices and **2** edges, with the exact edges listed out. If the graph has a [name] attribute, it is printed as well.
+
+Note
+
+`summary` is similar to `print` but does not list the edges, which is convenient for large graphs with millions of edges:
+
+```{r echo = TRUE}
+summary(gi)
+```
+
+## Adding/deleting vertices and edges
+
+Let's start from the empty graph again. To add vertices to an existing graph, use `add_vertices`:
+
+```{r}
+gi <- make_empty_graph()
+
+gi <- add_vertices(gi,3)
+```
+
+In igraph, vertices are always numbered up from one. The number of a vertex is called the *vertex ID*. A vertex might or might not have a name.
+
+Similarly, to add vertices use `add_edges`:
+
+```{r}
+gi <- add_edges(gi, edges = c(1,2, 1,3))
+```
+
+Edges are added by specifying the source and target vertex for each edge. This call added two edges, one connecting vertices `1` and `2`, and one connecting vertices `1` and `3`. Edges are also numbered up from one (the *edge ID*) and have an optional name.
+
+**Warning**
```

Comment:
I tried the `+` thing and it's kind of tricky because the plus operator does not automatically create vertices if they don't exist already. I.e. if the graph has 3 vertices and we do `g + edges(c(1,2, 3,4))`, it will throw an error. I understand the logic, but I feel like that's a little too soon in the tutorial to bother new users with this kind of baggage.

Nice idea though with the operator :-)

Reply:
@iosonofabio `add_edges()` behaves the same: it will not create new vertices.